lib/vue_phoenix/models/user.ex

```diff
@@ -32,11 +32,16 @@ defmodule VuePhoenix.User do
   end
 
   def sign_in(email, password) do
-    case Encryption.validate_password(Repo.get_by(User, email: email), password) do
-      {:ok, user} ->
-        token = Authenticator.generate_token(user)
-        Repo.insert(Ecto.build_assoc(user, :tokens, %{code: token}))
-      err -> err
+    case Repo.get_by(User, %{email: email}) do
+      nil -> {:error, %{email: "Your email address is not registered"}}
+      user ->
+        case Encryption.validate_password(user, password) do
+          {:ok, user} ->
+            token = Authenticator.generate_token(user)
+            Repo.insert(Ecto.build_assoc(user, :tokens, %{code: token}))
```

Comment:
Not require but this is the common suggested style 

```elixir
user
|> Ecto.build_assoc(:tokens, %{code: token})
|> Repo.insert
```

Reply:
Yeah, I forgot it. I'm still familiar with the syntax of rails